### PR TITLE
Gh12239 add exportdate exportby

### PR DESCRIPTION
**Pull request by riadmen**
Add new exportdate and exportby colums in table DatevAcctExport

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java [hunk 1]
```diff
@@ -56,6 +58,9 @@ protected String doIt()
 		final String sql = getSql();
 		final Evaluatee evalCtx = getEvalContext();
 
+		final int recordId = getRecord_ID();
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java:61]: you don't have to load this so early. you can do it when u are updating it (below)

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java [hunk 2]
```diff
@@ -92,6 +97,10 @@ else if (spreadsheetFormat == SpreadsheetFormat.CSV)
 			throw new AdempiereException("Unknown spreadsheet format: " + spreadsheetFormat);
 		}
 
+		datevAcctExport.setExportDate(SystemTime.asTimestamp());
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/impexp/spreadsheet/process/ExportToSpreadsheetProcess.java:100]: refactor this to a separate method called `updateDatevAcctExportIfNeeded` (or similar).

In that method you will:
* check if getTableName() is I_DatevAcctExport.Table_Name and if not, do nothing
* load the datevAcctExport record (as u did above), update it and save it (like you are doing it here)

File: backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620220_sys_gh12239_Add_Exportdate_Exportby.sql [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+;
+
+-- 2021-12-24T11:52:44.161Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+/* DDL */  select update_Column_Translation_From_AD_Element(580441) 
+;
+
+-- 2021-12-24T11:57:17.389Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Column SET AD_Reference_ID=11, DefaultValue='0', IsMandatory='Y',Updated=TO_TIMESTAMP('2021-12-24 12:57:17','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=579025
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620220_sys_gh12239_Add_Exportdate_Exportby.sql:83]: why have default value ZERO? IMHO it shall be empty and the field shall not be mandatory.

Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5620220_sys_gh12239_Add_Exportdate_Exportby.sql:83]: yes you are right Teo, but if you check the rest of the script this column has been created and replaced by ExportBy_ID

**Review by teosarca — APPROVED** [review]

**Pull request merged**